torch/_dynamo/guards.py

```diff
@@ -596,15 +596,16 @@ def guard_on_dict_keys_and_ignore_order(self, example_value, guard):
 
         # Iterate over the dicts and install a dict_getitem_manager.
         dict_source = guard.originating_source.name()
-        for key in example_value.keys():
+        for ind, key in enumerate(example_value.keys()):
+            key_source = ConstDictKeySource(guard.originating_source, ind)
```

Comment:
Dont do this. This will increase guard overhead. You need to take `guard_on_dict_keys_and_order` path here.